library/net.cpp

```diff
@@ -1410,3 +1418,274 @@ int dnet_trans_forward(struct dnet_io_req *r, struct dnet_net_state *orig, struc
+
+		r->st = dnet_state_get(st);
+		dnet_schedule_io(st->n, r);
+	};
+
+	n2_repliers repliers_wrappers;
+	repliers_wrappers.reply_error = [reply_error = std::move(repliers.reply_error), enqueue_response](int errc){
+		enqueue_response(std::bind(reply_error, errc));
+	};
+	repliers_wrappers.reply = [reply = std::move(repliers.reply),
+	                           enqueue_response](std::unique_ptr<n2_message> msg){
+		enqueue_response(n2_bind_noncopyable(reply, std::move(msg)));
+	};
+	return repliers_wrappers;
+}
+
+static int n2_trans_send(dnet_trans *t, n2_request_info *request_info) {
+	using namespace ioremap::elliptics;
+
+	struct dnet_net_state *st = t->st;
+	struct dnet_test_settings test_settings;
+	int err;
+
+	dnet_trans_get(t);
+
+	BOOST_SCOPE_EXIT(&t) {
+		dnet_trans_put(t);
+	} BOOST_SCOPE_EXIT_END
+
+	pthread_mutex_lock(&st->trans_lock);
+	err = dnet_trans_insert_nolock(st, t);
+	if (!err) {
+		dnet_trans_update_timestamp(t);
+		dnet_trans_insert_timer_nolock(st, t);
+	}
+	pthread_mutex_unlock(&st->trans_lock);
+	if (err)
+		return err;
+
+	if (t->n->test_settings && !dnet_node_get_test_settings(t->n, &test_settings) &&
+	    test_settings.commands_mask & (1 << t->command))
+		return err;
+
+	auto repliers_wrappers = n2_make_repliers_via_request_queue(st,
+	                                                            request_info->request->cmd,
+	                                                            std::move(request_info->repliers));
+
+	n2::net_state_get_protocol(st)->send_request(st,
+	                                             std::move(request_info->request),
+	                                             std::move(repliers_wrappers));
+
+	return err;
+}
+
+static int n2_trans_complete_forward_impl(struct dnet_addr * /*addr*/, n2_response_info *response_info, void *priv) {
+	auto cmd = &response_info->cmd;
+
+	if (!is_trans_destroyed(cmd)) {
+		response_info->response_holder();
+		return 0;
+	}
+
+	return -EINVAL;
+}
+
+static int n2_trans_complete_forward(struct dnet_addr *addr, n2_response_info *response_info, void *priv) {
+	auto t = static_cast<dnet_trans *>(priv);
+	return c_exception_guard(std::bind(&n2_trans_complete_forward_impl, addr, response_info, priv),
+	                         t->st->n, __FUNCTION__);
+}
+
+int n2_trans_forward(n2_request_info *request_info, struct dnet_net_state *orig, struct dnet_net_state *forward) {
+	dnet_cmd *cmd = &request_info->request->cmd;
+
+	auto t = dnet_trans_alloc(orig->n, 0);
+	if (!t)
+		return -ENOMEM;
+
+	t->rcv_trans = cmd->trans;
+	cmd->trans = t->cmd.trans = t->trans = atomic_inc(&orig->n->trans);
+
+	t->cmd = *cmd;
+
+	t->wait_ts = [&cmd, &t, &request_info]() -> timespec {
+		using namespace ioremap::elliptics;
+
+		auto deadline = request_info->request->deadline;
+
+		if (dnet_time_is_empty(&deadline))
+			return t->wait_ts;
+		else
+			return dnet_time_left_to_timeout(deadline);
+	}();
+
+	if (!t->wait_ts.tv_sec && !t->wait_ts.tv_nsec) {
```

Comment:
just a note: here you use curly braces in a single line if. That looks great for me, but I'm unsure which style do we use

Reply:
There is mixed style in the project. The main guide is to have the same style (about one-line code blocks) within the function or module.